deployments/log_analysis.yml

```
@@ -871,3 +863,24 @@ Resources:
       FunctionName: !Ref RulesEngineFunction
       FunctionTimeoutSec: !FindInMap [Functions, RulesEngine, Timeout]
       ServiceToken: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-cfn-custom-resources
+
+  RulesEngineLayers:
+    Type: Custom::LayerAttachment
+    Properties:
+      FunctionArn: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-rules-engine
+      LayerArns: !If
+        - AttachLayers
+        - !Split # CFN doesn't have list append, so convert to/from CSV string to prepend base layers
+          - ','
+          - !Sub
+            - '${base},${pip},${global}'
+            - base: !Join [',', !Ref LayerVersionArns]
+              pip: !Ref PythonLayerVersionArn
+              global: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:layer:panther-engine-globals:LATEST
+        - !Split # CFN doesn't have list append, so convert to/from CSV string to prepend base layers
+          - ','
+          - !Sub
+            - '${pip},${global}'
+            - pip: !Ref PythonLayerVersionArn
+              global: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:layer:panther-engine-globals:LATEST
```

Damn, CloudFormation makes this ugly. We should eventually change the "LayerVersionArns" parameter to just accept a single layer, users can't and won't realistically add multiple anyway. Then we don't have to do this split + join hack.

In the meantime, lets move the conditional deeper to avoid having to duplicate the entire list:

```suggestion
      # CFN doesn't have list append, so convert to/from CSV string to prepend base layers
      LayerArns: !Split
        - ','
        - !Sub
          - '${base}${comma}${pip},${global}'
          - base: !Join [',', !Ref LayerVersionArns]
            comma: !If [AttachLayers, ',', ''] # add comma after base layers if applicable
            pip: !Ref PythonLayerVersionArn
            global: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:layer:panther-engine-globals:LATEST
```

same for the policy engine. I tested this (without the global) and it worked